src/main/java/org/jboss/modules/Module.java

```diff
@@ -113,7 +114,7 @@ public final class Module {
         AccessController.doPrivileged(new PrivilegedAction<Void>() {
             public Void run() {
                 try {
-                    URL.setURLStreamHandlerFactory(ModularURLStreamHandlerFactory.INSTANCE);
+                    URL.setURLStreamHandlerFactory(ModularURLStreamHandlerProvider.INSTANCE);
```

Comment:
Maybe we should eliminate this (and the similar block below), and replace it with `provides URLStreamHandlerProvider with ModularURLStreamHandlerProvider` (and corresponding `META-INF/services` files) in the module-info? It would require these classes being made public, however we could move them to the non-exported `_private` subpackage in this case.

The advantage of doing this is that it would be one less step for the bootstrap API to deal with. However we could make this change in a follow-up pull request after more discussion if needed.

Reply:
I created https://issues.redhat.com/browse/MODULES-451 as follow up of this proposal.